sympy/physics/mechanics/kane.py

```diff
@@ -55,6 +57,17 @@ class KanesMethod(_Methods):
         Boolean whether the mass matrices and forcing vectors should use the
         explicit form (default) or implicit form for kinematics.
         See the notes for more details.
+    kd_eqs_solver : str, callable
+        Solver to be used for the kinematic differential equations. The default
+        is LU solve. See the notes for more details on what the solver is used
+        for and advise on what solver should be used.
+    constraint_solver : str, callable
+        Solver to be used for the velocity constraints. The default is LU solve.
+        See the notes for more details on what the solver is used for and advise
+        on what solver should be used.
+    use_block_matrices : bool
+        Boolean whether the mass matrices and forcing vector should be returned
+        as block matrices.
```

Comment:
It isn't clear why this is needed or what it does.

Reply:
This is needed for `MatrixSolve` as explained in the notes, but I'll try to clarify it a bit further.